src/ram/IndexChoice.h

```diff
@@ -53,13 +53,12 @@ namespace souffle::ram {
  */
 class IndexChoice : public IndexOperation, public AbstractChoice {
 public:
-    IndexChoice(Own<RelationReference> r, int ident, Own<Condition> cond, RamPattern queryPattern,
+    IndexChoice(const std::string& rel, int ident, Own<Condition> cond, RamPattern queryPattern,
```

Comment:
This should be just a string, since we end up copying it anyway. I think this is missed by clang-tidy since it gets passed up through a few constructors. The same applies for the other ram constructors that now use `std::string`.
```suggestion
    IndexChoice(std::string rel, int ident, Own<Condition> cond, RamPattern queryPattern,
```

Reply:
Done.